test/Catch.hs

```diff
@@ -11,26 +11,24 @@ import Test.Tasty.Hedgehog
 
 gen
   :: forall e m a sig
-  .  (Has (Catch e) sig m, Arg e, Show a, Show e, Vary e)
+  .  (Has (Catch e) sig m, Arg e, Show e, Vary e)
   => Gen e
-  -> (forall a . Show a => Gen a -> Gen (With (m a)))
+  -> (forall a . Gen a -> Gen (m a))
   -> Gen a
-  -> Gen (With (m a))
-gen _ m a = do
-  h <- fn @e (m a)
-  subterm (m a) $ \ m -> liftWith2 "catchError" catchError m (fmap getWith <$> showingFn h)
+  -> Gen (m a)
+gen _ m a = label "catchError" catchError <*> m a <*> fn @e (m a)
```

Comment:
We now have much simpler combinators for generating showable-by-construction terms:

- `atom` & `label` construct prefix terms
- `infixL` & `infixR` construct infix terms
- `<*>` does function application (as ever)